### DISCUSSION TITLE
Open Nvim Tree on startup while preserving vim-startify

### DISCUSSION THREAD

**Discussion by unayakani** [nvim-tree/nvim-tree.lua#1771] [Q&A] [answered]
***How do I open nvim tree on startup along with vim-startify?***

**I use Neovide**

I tried this:
```vim
autocmd VimEnter *
            \   if !argc()
            \ |   Startify
            \ |   NvimTreeOpen
            \ |   wincmd w
            \ | endif
```
But it doesn’t seem to work. Isn’t this a simple question and shouldn’t it be easy to find? I read the nvim tree docs but no luck. I’ve been trying from hours. I also saw this [issue](https://github.com/nvim-tree/nvim-tree.lua/discussions/1517) but I haven’t been able to open startify along with nvim tree on startup. Thankyou.  
(And also, I wanted to know how to make nvim tree startup on a specific directory after we have solved the above issue. Thankyou)

**Comment by unayakani** [comment]
Quick update.:
It just clicked to me right now, so now I have **temporary** fix:
```vim
autocmd VimEnter * :Startify
autocmd VimEnter * :NvimTreeOpen
```
I still do **need a permanent thing** to put in the nvim-tree.lua config

**Comment by unayakani** [comment]
Now I gotta figure out how to throw the cursor to the other window coz `| wincmd p` isn't working

**Comment by alex-courtis** [comment]
> ```viml
> autocmd VimEnter * :NvimTreeOpen
> ```

It might be easier to use the option `:help nvim-tree.open_on_setup` to do this for you.

> Now I gotta figure out how to throw the cursor to the other window coz | wincmd p isn't working

See `:help nvim-tree.open_on_setup_file`

**Reply by unayakani** [reply]
Open on setup file didn't work earlier. I'll got through the help files for sure. Thanks.

**Reply by unayakani** [reply]
`:h nvim-tree.open_on_setup_file` states that "nvim-tree will be focused". I didn't want that.

**Reply by unayakani** [reply]
Update: open on setup and open on setup file; both aren't working

**Reply by unayakani** [reply]
Adding `open_on_setup = true` _**and**_ `ignore_buffer_on_setup = true` did the job!!

**Comment by unayakani** [comment, marked as answer]
@alex-courtis was partially correct,
Only `open_on_setup = true` didn't work for vim-startify.
I also had to add `ignore_buffer_on_setup = true` after that to do the trick.
Here's a part of my new config:
```vim
lua << EOF
require("nvim-tree").setup{
  open_on_setup = true,
  ignore_buffer_on_setup = true,
}
EOF
```
( I still use VimScript 😂 )

**Reply by unayakani** [reply]
Thanks again @alex-courtis . I had seen the open on setup things before but adding ignore buffer on setup does the trick!

**Reply by alex-courtis** [reply]
> ( I still use VimScript  )

Nothing wrong with that... I'm still using init.vim

It calls lua files when lua is needed for setup etc: https://github.com/alex-courtis/arch/blob/6d681e76b52c7123ce53d18c5fd107cab6739d0a/config/nvim/init.vim#L100

**Reply by unayakani** [reply]
> Nothing wrong with that

I know. Even I call Lua files either with `lua << EOF` or for single line configs, `lua`
Just that people seem to be lua-izing everything and I’m just fine with Vimscript. I find vimscript to be easier, too.

**Comment by unayakani** [comment]
@alex-courtis , can you add this solution as a section in the wiki? Looks like forking the repo doesn’t fork the wiki and I have to clone it manually.

**Reply by alex-courtis** [reply]
Unfortunately there doesn't appear to be a means to open up the wiki repo for direct pushing.

Fortunately, everyone can just edit the wiki directly :)
